supervise/main.go

```diff
@@ -78,31 +82,35 @@ MSGID = The message "type". Set to "out" or "err" to specify which stdio
 		fatalf("reporting pid: %v", err)
 	}
 
+	// Dial syslog.
+	conn, err := net.DialUDP("udp", nil, udpAddr)
+	if err != nil {
+		fatalf("dialing udp: %w", err)
+	}
+	defer conn.Close()
+
 	// Handle signals.
 	c := make(chan os.Signal)
-	signal.Notify(c, os.Interrupt, os.Kill, syscall.SIGCHLD)
+	signal.Notify(c, os.Interrupt, syscall.SIGTERM, syscall.SIGCHLD)
 	go func() {
 		for sig := range c {
 			switch sig {
 			// Forward signals to child.
-			case os.Interrupt, os.Kill:
-				if err := cmd.Process.Signal(sig); err != nil {
+			case os.Interrupt, syscall.SIGTERM:
+				if err := cmd.Process.Signal(syscall.SIGTERM); err != nil {
 					break
 				}
+				// After some timeout, check if the process is still running and, if so,
```

Comment:
This comment suggests some checks/conditions, but it seems to um.. be unconditionally :-P Tho I think unconditional is fine here, assuming no immediate pid reuse (and I think there is nothing better you can do beyond that).

Reply:
Fixed — I removed the check when I was debugging and forgot to add it back.